### PR TITLE
feat(viz): visualize grid

### LINKED ISSUE #446: make_grid visualization function for images
https://pytorch.org/vision/stable/generated/torchvision.utils.make_grid.html

This is a great utility function.  I have something similar in a tutorial:

```
def visualize_dataset(dataset, title):
    plt.figure(figsize=(6, 6)).suptitle(title, fontsize=18)
    for i, samples in enumerate(iter(dataset.take(9))):
        images = samples["images"]
        plt.subplot(3, 3, i + 1)
        plt.imshow(images[0].numpy().astype("uint8"))
        plt.axis("off")
    plt.show()
```

### PR DESCRIPTION
**Pull request by kartik4949**
closes #446 
This is initial setup of this visualisation

### CODE REVIEW

File: keras_cv/visualization/utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import matplotlib.pyplot as plt
+
+def visualize_grid(ds, figsize=(8, 8)):
```
[the remark this answers is not in the document]
Comment (innat) [reply keras_cv/visualization/utils.py:17]: @kartik4949 I think both types should be able to pass.

File: keras_cv/visualization/utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     images, _ = next(iter(ds.take(1)))
```
Comment (innat) [inline keras_cv/visualization/utils.py:18]: In the case of visualizing an unsupervised dataset, can we add support for that?

Comment (innat) [reply keras_cv/visualization/utils.py:18]: We can use a boolean positional argument, `title` . For example 

```
if title:
   plt.title(...)
```

File: keras_cv/visualization/utils.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+     plt.figure(figsize=figsize)
+     for i in range(9):
+         plt.subplot(3, 3, i + 1)
+         plt.imshow(images[i].numpy().astype("uint8"))
+         plt.axis("off")
+     plt.show()
```
Comment (innat) [inline keras_cv/visualization/utils.py:24]: Can we add `title_layout`? 
```
plt.tight_layout()
plt.show()
```

**Comment by LukeWood** [comment]
Let’s perfect our visualization tools in the demo utils file to start out!  Then we can move on to a standard api

**Comment by LukeWood** [comment]
Hey @kartik4949 !  I really appreciate your contributions to KerasCV.  We have decided to NOT branch out into visualization tools just yet.  If we do, I'd like to contribute the first component myself so I can think carefully about the API and make those decisions in a vacuum.  Sorry about that, and I hope you contribute again in the future.

**Pull request closed without merging**